### PR TITLE
Update etcd

### PR DESCRIPTION
**Pull request by AkihiroSuda**
This PR updates etcd, but it seems ~~constantly~~ intermittently hitting #93  (https://github.com/coreos/etcd/commit/6e290abee217b222178051cc8f5ea6f163cdf1bd).

cc @siddontang 

[a.txt](https://github.com/aphyr/jepsen/files/372880/a.txt):
```
lein test :only jepsen.etcd-test/register-test

FAIL in (register-test) (etcd_test.clj:46)
expected: (:valid? (:results test))
  actual: false
Not linearizable. Linearizable prefix was:

Followed by inconsistent operation:
                nil     nil

-------------------------------------------------------------
Just prior to that operation, possible interpretations of the
linearizable prefix were:

lein test :only jepsen.etcd-test/register-test

ERROR in (register-test) (ArrayList.java:177)
Uncaught exception, not in assertion.
expected: nil
  actual: java.lang.NullPointerException: null
 at java.util.ArrayList.<init> (ArrayList.java:177)
    clojure.core$shuffle.invoke (core.clj:6981)
    jepsen.report$linearizability.invoke (report.clj:36)
    jepsen.etcd_test/fn (etcd_test.clj:47)
    clojure.test$test_var$fn__7670.invoke (test.clj:704)
    clojure.test$test_var.invoke (test.clj:704)
    clojure.test$test_vars$fn__7692$fn__7697.invoke (test.clj:722)
    clojure.test$default_fixture.invoke (test.clj:674)
    clojure.test$test_vars$fn__7692.invoke (test.clj:722)
    clojure.test$default_fixture.invoke (test.clj:674)
    clojure.test$test_vars.invoke (test.clj:718)
    clojure.test$test_all_vars.invoke (test.clj:728)
    clojure.test$test_ns.invoke (test.clj:747)
    clojure.core$map$fn__4553.invoke (core.clj:2624)
    clojure.lang.LazySeq.sval (LazySeq.java:40)
    clojure.lang.LazySeq.seq (LazySeq.java:49)
    clojure.lang.Cons.next (Cons.java:39)
    clojure.lang.RT.boundedLength (RT.java:1735)
    clojure.lang.RestFn.applyTo (RestFn.java:130)
    clojure.core$apply.invoke (core.clj:632)
    clojure.test$run_tests.doInvoke (test.clj:762)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invoke (core.clj:630)
    user$eval85$fn__144$fn__175.invoke (form-init4066669656748077365.clj:1)
    user$eval85$fn__144$fn__145.invoke (form-init4066669656748077365.clj:1)
    user$eval85$fn__144.invoke (form-init4066669656748077365.clj:1)
    user$eval85.invoke (form-init4066669656748077365.clj:1)
    clojure.lang.Compiler.eval (Compiler.java:6782)
    clojure.lang.Compiler.eval (Compiler.java:6772)
    clojure.lang.Compiler.load (Compiler.java:7227)
    clojure.lang.Compiler.loadFile (Compiler.java:7165)
    clojure.main$load_script.invoke (main.clj:275)
    clojure.main$init_opt.invoke (main.clj:280)
    clojure.main$initialize.invoke (main.clj:308)
    clojure.main$null_opt.invoke (main.clj:343)
    clojure.main$main.doInvoke (main.clj:421)
    clojure.lang.RestFn.invoke (RestFn.java:421)
    clojure.lang.Var.invoke (Var.java:383)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.main.main (main.java:37)

Ran 1 tests containing 2 assertions.
1 failures, 1 errors.
Tests failed
```

ZIP: 
[20160720T040716.000Z.zip](https://github.com/aphyr/jepsen/files/372890/20160720T040716.000Z.zip)

### CODE REVIEW

File: etcd/test/jepsen/etcd_test.clj [hunk 1]
```diff
@@ -24,7 +24,7 @@
                  :db        (db)
                  :client    (cas-client)
                  :model     (model/cas-register)
-                 :checker   (checker/compose {:html   timeline/html
+                 :checker   (checker/compose {;:html   timeline/html
```
Comment (AkihiroSuda) [inline etcd/test/jepsen/etcd_test.clj]: This is just a "work-around" for `java.lang.IllegalArgumentException: No implementation of method: :check of protocol: #'jepsen.checker/Checker found for class: jepsen.checker.timeline$html`

**Comment by siddontang** [comment]
Hi @AkihiroSuda 
This means that the newest etcd still has this problem?

**Comment by AkihiroSuda** [comment]
@siddontang yes, but I'm not sure this is a false-positive or a real linearization bug

**Comment by AkihiroSuda** [comment]
Uh, I did not set `:quorum?`. I'll try it later :sweat_smile:

**Comment by AkihiroSuda** [comment]
@siddontang Even with `:quorum?`, it seems hitting #93 

Evidence on travis: https://travis-ci.org/AkihiroSuda/jepsen/builds/146052784 (commit https://github.com/AkihiroSuda/jepsen/commit/4994e76a40ce5eafbb486ad13fb038aa2dd03a5c)

**Comment by aphyr** [comment]
@AkihiroSuda: So, the way you tell if this is a Jepsen bug or an etcd bug is to go and stare at the history for a while, and see if you can convince yourself the state is nonlinearizable. I don't... *think* we've had a false nonlinearizable result in Jepsen since the very early days of Knossos, but it might be worth upgrading to 0.1.2 just in case.

**Comment by aphyr** [comment]
screenshot from 2016-07-20 10 10 04

Judging by the linear.svg output, this looks nonlinearizable. It looks like there's some sort of crash in the test analyzer though, so I'd get that sorted out first. Can't remember how 0.0.7 did things, exactly.

File: etcd/test/jepsen/etcd_test.clj (hunk 1 above)
Comment (aphyr) [inline etcd/test/jepsen/etcd_test.clj]: Ah, I think it's (timeline/html) now--a function. :)

**Comment by aphyr** [comment]
Ah, yeah, I'm out of date and catching up here: you are on 0.1.2. Lemme merge and take a look at that shuffle bug.

**Comment by aphyr** [comment]
> Even with :quorum?

So this PR uses verschlimmbersserung 0.1.1, which didn't support `:quorum?`. I merged a PR a while back for it but didn't release. Cutting a release of that and testing now.

**Comment by aphyr** [comment]
The error at the end comes about because report/linearizability is deprecated--the new checker emits one automatically. Dropped that.

**Comment by aphyr** [comment]
Re-tested with :quorum? true and it looks like it passes. At least, using the same schedule that fails with :consistent? true, haha. :)

**Comment by AkihiroSuda** [comment]
Great, thank you a lot for looking into this

I should have carefully checked versch :sweat:

**Pull request merged**